### PR TITLE
Fix Opal on a C++ client

### PR DESCRIPTION
**Pull request by theiostream**
These two commits allow Opal to work on a C++ context.

### CODE REVIEW

**Comment by theiostream** [comment]
@ivucica

**Comment by ivucica** [comment]
Second commit makes sense, for the first one I'm roping in @ericwa again

**Comment by theiostream** [comment]
Just a brief comment then: I am far from being certain that my solution to this issue is the best; but in C++, `CGPathRef x = CGPathCreateMutable();` would error; which is not API-compatible. This needs to be addressed _somehow_.

File: Headers/CoreGraphics/CGPath.h [hunk 1]
```diff
@@ -42,16 +42,11 @@ typedef enum CGPathElementType {
...
-typedef struct CGPath* CGPathRef;
-#endif
-
-#ifdef INTERNAL_BUILD_OBJC
-@class CGMutablePath;
-typedef CGMutablePath* CGMutablePathRef;
-#else
-typedef struct CGMutablePath* CGMutablePathRef;
+typedef const struct CGPath* CGPathRef;
+typedef struct CGPath* CGMutablePathRef;
```
Comment (ericwa) [inline Headers/CoreGraphics/CGPath.h]: Couldn't we keep the CGMutablePath/CGPath classes as-is internally, but still make the public API change, i.e. this part:?
```
#ifdef INTERNAL_BUILD_OBJC
...
#else
typedef const struct CGPath* CGPathRef;
typedef struct CGPath* CGMutablePathRef;
#endif
```

Comment (theiostream) [reply Headers/CoreGraphics/CGPath.h]: Theoretically, yes. Stupid of me not to think of it. I'll see if anything goes wrong with that approach.

**Comment by theiostream** [comment]
@ericwa @ivucica I grouped some more header API compatibility fixes in this pull request. Could you comment?

Thank you for the feedback so far.

**Comment by ericwa** [comment]
Looks good to me.

```
enum {
...
} CTFontSymbolicTraits;
typedef int CTFontSymbolicTraits;
```
seems to be the standard way enums are done in libs-gui.

**Comment by theiostream** [comment]
I suppose you meant:

```
enum {
...
};
```

, without the last `CTFontSymbolicTraits`, which is what this pull request does? Otherwise you're declaring a global variable called `CTFontSymbolicTraits`.

**Comment by ericwa** [comment]
Sorry, typo on my part. I just meant that what you have in the pull request looks good.

**Comment by theiostream** [comment]
Thanks. Waiting for final feedback from @ivucica then :)

**Review by fredkiefer — COMMENTED** [review]
Looks good to me.

**Comment by ivucica** [comment]
I see no problems upon a (quick) inspection. Merging.

**Pull request merged**